index.js

```diff
@@ -136,49 +191,61 @@ function nowServing(line) {
 }
 
 const line = []
+let number = 0
 
-// ***** Scope & Closures - Question 1 *****
+ // ***** Scope & Closures - Question 1 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+ console.log("%cScope & Closures - Question 1", "color: red")
+
+ function takeATicketNumber (line) {
+  number = number + 1
+  line.push(number)
+  return `Welcome. You are ticket number ${number}`
+ }
 
-// console.log(takeATicketNumber(line))
+ console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+ console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+ console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+ console.log(nowServing(line))
 // // => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
+ console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+ console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
+
+
+ console.log("%cScope & Closures - Question 2", "color: red")
+
+ //function TakeATicketNumberFunc () {}
```

Comment:
I would take a look at the solution in the `solution` branch!
It's an interesting solution!